### PR TITLE
added communities browse page. part of #76 and #248

### PR DESCRIPTION
**Pull request by weiweishi**
A Community browsing page is needed for the launch. 
also needed a flag property to easily distinguish community and collection in addition to the specifications in #569
added select_collections_controller methods for selecting communities/collections as needed.

### CODE REVIEW

**Comment by weiweishi** [comment]
I'm breaking my initial work on #92 into smaller and sensible chunks to avoid confusion during debugging. Hope this makes sense.

**Comment by weiweishi** [comment]
and @nnunn this should give you what you need for the UI work on browsing the communities. Let me know if you need any help from me.

**Comment by nnunn** [comment]
Great @weiweishi - I am currently working through the ccid ui changes. Once that's done I will tackle this. I may have some questions then.

File: spec/controllers/selects_collections_spec.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+class SelectsCollectionsController < ApplicationController
+  include Blacklight::Catalog
+  include Hydranorth::Collections::SelectsCollections
+  include Hydra::Controller::ControllerBehavior
+  include Hydranorth::Permissions
```
Comment (pgwillia) [inline spec/controllers/selects_collections_spec.rb]: <class:SelectsCollectionsController>': uninitialized constant Hydranorth::Permissions (NameError)

Comment (weiweishi) [inline spec/controllers/selects_collections_spec.rb]: forgot to commit those related files. will Fix it.

Comment (weiweishi) [inline spec/controllers/selects_collections_spec.rb]: removed this dependency as it doesn't belong to this PR.

**Comment by pgwillia** [comment]
@weiweishi I'm assuming you want to wait to merge this until @nnunn has styled it?

image

**Comment by nnunn** [comment]
This is styled. It would be really good to see it with production data though.

**Comment by weiweishi** [comment]
it looks like this now:
image

Can we add some whitespaces between boxes, and maybe make the font smaller so it will fit the full title? @nnunn @sfbetz

**Comment by mbarnett** [comment]
Code looks fine to me. I'm assuming this is waiting on style fixes?

**Comment by weiweishi** [comment]
not sure if @nnunn is still around to fix the style. I'll see if I can make
the change.

*Weiwei Shi*

Digital Initiative Applications Librarian
University of Alberta Libraries
2-10L Cameron Library
Edmonton, Alberta, Canada T6G 2J8
Phone:(780)492-7802
Fax: (780)248-1209
Email: weiwei.shi@ualberta.ca

**Comment by pbinkley** [comment]
If not, let's merge as is and fix the style post-launch.

**Comment by weiweishi** [comment]
changed it to:
image

**Comment by sfbetz** [comment]
Looks better. Thanks Weiwei!


On Wed, Sep 30, 2015 at 5:04 PM, Weiwei Shi <notifications@github.com>
wrote:

> changed it to:
> [image: image]
> <https://cloud.githubusercontent.com/assets/2047614/10209072/40e1a3a6-6795-11e5-8c82-6e2eaad90a34.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/pull/637#issuecomment-144570007>
> .
>



-- 
Sonya Betz
Digital Initiatives Projects Librarian
Acting Digital Repository <http://era.library.ualberta.ca/> Services
Librarian



*University of Alberta Libraries2-10U Cameron Library, University of
AlbertaEdmonton, AB, Canada T6G 2J8  *
*sonya.betz@ualberta.ca <sfbetz@ualberta.ca> *
*780-492-1718*

**Comment by pbinkley** [comment]
Looks great. Print it!

(This is where we need to those manual-typewriter USB keyboards, and an app
that merges the PR you're looking at when you yell "COPY!")

**Comment by nnunn** [comment]
Thanks @weiweishi - sorry I had left already.

**Comment by weiweishi** [comment]
aside from the migration test, which should be fixed by #646 

Tests failed include:


  1) Hydranorth::GenericFile::Actor#update_metadata should update year_created based on date_created
     Failure/Error: actor.update_metadata({date_created:'2012/01/01'}, 'open')
     NoMethodError:
       undefined method `empty?' for nil:NilClass
     # ./app/actors/hydranorth/generic_file/actor.rb:21:in `update_metadata'
     # ./spec/actors/hydranorth/generic_files/actor_spec.rb:14:in `block (3 levels) in <top (required)>'

  2) coauthor abby shares work @ included
     Failure/Error: click_link "Permissions"
     Capybara::ElementNotFound:
       Unable to find link "Permissions"
     # ./spec/features/coauthor_spec.rb:73:in `share_work_with'
     # ./spec/features/coauthor_spec.rb:34:in `block (3 levels) in <top (required)>'

  3) coauthor abby shares work @ not included
     Failure/Error: click_link "Permissions"
     Capybara::ElementNotFound:
       Unable to find link "Permissions"
     # ./spec/features/coauthor_spec.rb:73:in `share_work_with'
     # ./spec/features/coauthor_spec.rb:38:in `block (3 levels) in <top (required)>'

  4) coauthor abby shares work barbara can edit and feature
     Failure/Error: click_link "Permissions"
     Capybara::ElementNotFound:
       Unable to find link "Permissions"
     # ./spec/features/coauthor_spec.rb:73:in `share_work_with'
     # ./spec/features/coauthor_spec.rb:50:in `block (3 levels) in <top (required)>'

  5) Hydranorth::Forms::GenericFileEditForm#terms should return a list
     Failure/Error: expect(subject.terms).to eq([:title, :creator, :contributor, :subject, :resource_type, :language, :identifier, :spatial, :temporal, :description, :date_created, :license, :rights, :is_version_of, :source, :related_url ])

       expected: [:title, :creator, :contributor, :subject, :resource_type, :language, :identifier, :spatial, :temporal, :description, :date_created, :license, :rights, :is_version_of, :source, :related_url]
            got: [:title, :creator, :contributor, :subject, :resource_type, :language, :identifier, :spatial, :temporal, :description, :date_created, :license, :rights, :is_version_of, :source, :related_url, :belongsToCommunity, :hasCollectionId]

File: app/controllers/browse_controller.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ def self.modified_field
     #   The ordering of the field names is the order of the display
     config.add_facet_field solr_name("resource_type", :facetable), label: "Item Type", limit: 20
     config.add_facet_field solr_name("subject", :facetable), label: "Subject", limit: 20
-    config.add_facet_field solr_name("hasCollection", :symbol), label: "Collection", limit: 20 
+    #config.add_facet_field solr_name("hasCollection", :symbol), label: "Collection", limit: 20
```
Comment (pgwillia) [inline app/controllers/browse_controller.rb]: I'm not a fan of changes like this.  It's either in or out, commented out code just make it harder to maintain.  Does this mean that the link "Browse collections" from the front page should be changed as well?

Comment (mbarnett) [inline app/controllers/browse_controller.rb]: Yeah, I agree. Kill dead code and trust your version control if you ever need it back. Commented out code is just a source of confusion.

**Comment by pgwillia** [comment]
This looks great fullscreen.  Not as nice in my half screen window.

image

File: app/controllers/browse_controller.rb (hunk 2 above)
Comment (weiweishi) [inline app/controllers/browse_controller.rb]: @nnunn has temporarily disabled the browsing by collection with @sfbetz's request. We will need sometime to sort out browsing by collection/community after the launch. But I'm okay with removing the line.

**Comment by pgwillia** [comment]
This is how I changed it:
image

**Comment by pgwillia** [comment]
I think this can be merged.  I've opened a separate issue for the failing tests and we can fix the style issue in the future.

**Comment by nnunn** [comment]
gahh yeah some the responsive stuff is breaking on new code. I will have that fixed by the new release on friday next week.

**Comment by pbinkley** [comment]
OK, I'm merging.
